src/platform/plugins/shared/discover/public/embeddable/components/search_embeddable_grid_component.tsx

```diff
@@ -96,9 +99,10 @@ export function SearchEmbeddableGridComponent({
 
   const isEsql = useMemo(() => isEsqlMode(savedSearch), [savedSearch]);
 
-  const sort = useMemo(() => {
-    return getSortForEmbeddable(savedSearch.sort, dataView, discoverServices.uiSettings, isEsql);
-  }, [savedSearch.sort, dataView, isEsql, discoverServices.uiSettings]);
+  const sort = useMemo(
+    () => getSortArray(savedSearch.sort ?? [], dataView, isEsql),
+    [dataView, isEsql, savedSearch.sort]
+  );
```

Comment:
This is the Discover embeddable bug. We were applying default time field sorting logic here that was only in the UI and not actually applied to the search (seems like it was in the past, but hasn't been for some time). I was on the fence about if I should update the fetching logic to actually use `getSortForEmbeddable` instead, but I believe removing it entirely is the correct behaviour.

If `getSortForEmbeddable` was applied to the actual search, any Discover session saved without a `sort` would automatically have default time field sorting applied when added to a dashboard, which I don't think reflects the user's intent. This is likely a rare situation anyway, but it probably makes sense to respect the user's configuration in the search and the UI.